core/tee/tee_svc_cryp.c

```diff
@@ -3097,28 +3118,36 @@ TEE_Result syscall_cipher_init(unsigned long state, const void *iv,
 
 	key1 = o->attr;
 
+	res = bb_memdup_user(iv, iv_len, &iv_bbuf);
+	if (res)
+		return res;
+
 	if (tee_obj_get(utc, cs->key2, &o) == TEE_SUCCESS) {
 		struct tee_cryp_obj_secret *key2 = o->attr;
 
-		if ((o->info.handleFlags & TEE_HANDLE_FLAG_INITIALIZED) == 0)
-			return TEE_ERROR_BAD_PARAMETERS;
+		if ((o->info.handleFlags & TEE_HANDLE_FLAG_INITIALIZED) == 0) {
+			res = TEE_ERROR_BAD_PARAMETERS;
+			goto out;
+		}
 
 		res = crypto_cipher_init(cs->ctx, cs->mode,
 					 (uint8_t *)(key1 + 1), key1->key_size,
 					 (uint8_t *)(key2 + 1), key2->key_size,
-					 iv, iv_len);
+					 iv_bbuf, iv_len);
 	} else {
 		res = crypto_cipher_init(cs->ctx, cs->mode,
 					 (uint8_t *)(key1 + 1), key1->key_size,
-					 NULL, 0, iv, iv_len);
+					 NULL, 0, iv_bbuf, iv_len);
 	}
 	if (res != TEE_SUCCESS)
-		return res;
+		goto out;
 
 	cs->ctx_finalize = crypto_cipher_final;
 	cs->state = CRYP_STATE_INITIALIZED;
 
-	return TEE_SUCCESS;
+out:
+	bb_free(iv_bbuf, iv_len);
```

Comment:
There is normally no need to free bounce buffers at the end of a `syscall_*()` function since the next thing to happen is that the bounce buffers are reset. In fact, I believe it's better to skip freeing bounce buffers in `syscall_*()` functions unless they are allocated in a loop or in other ways may accumulate.

Please consider this pattern in all the other syscall functions here too.

Reply:
I see. Removed `bb_free()` from the epilogue of system call handlers